services/searchengine/bleveengine/search.go

```diff
@@ -708,7 +716,25 @@ func (b *BleveEngine) SearchFiles(channels model.ChannelList, searchParams []*mo
 
 		if params.Terms != "" {
 			terms := []string{}
-			for _, term := range strings.Split(params.Terms, " ") {
+
+			// Since we will put wildcard on the rest of the terms,
+			// we will get exactPhraseTerms first and then remove the matching terms
+			exactPhraseTerms := exactPhraseRegExpForFileSearch.FindAllString(params.Terms, -1)
+			params.Terms = exactPhraseRegExpForFileSearch.ReplaceAllLiteralString(params.Terms, "")
+
+			wildcardAddedTerms := strings.Fields(params.Terms)
+
+			// A wildcard is attached to the end of every word
+			// regardless of whether or not a wildcard exists
+			for index, term := range wildcardAddedTerms {
+				if !strings.HasPrefix(term, "*") {
+					wildcardAddedTerms[index] = wildcardRegExpForFileSearch.ReplaceAllLiteralString(term, "*")
+				}
+			}
```

Comment:
Sorry, but I don't understand this: Why do we check if the term starts with an asterisk and skip in that case?

Reply:
Also, is this the same as the following?

```go
// A wildcard is attached to the end of every word
// regardless of whether or not a wildcard exists
for index, term := range wildcardAddedTerms {
	if !strings.HasPrefix(term, "*") && !strings.HasSuffix(term, "*") {
		wildcardAddedTerms[index] = term + "*"
	}
}
```